insonmnia/worker/server.go

```diff
@@ -365,6 +373,17 @@ func (m *Worker) setupPlugins() error {
 	return nil
 }
 
+func (m *Worker) setupMetrics() error {
+	h, err := metrics.NewHandler(m.ctx, m.cfg.Plugins.GPUs)
```

Comment:
Can we avoid passing context in New* functions, like here?

Reply:
But we must terminate the metrics plugin on worker exit. Plugin stops collecting metrics when this context is canceling, like a lot of other components.